### DISCUSSION TITLE
IP Address MongoDB server connection

### DISCUSSION THREAD

**Discussion by slayt12** [vercel/community#3981] [Help]
### Summary

My app uses MongoDB for its backend, and when I whitelist the IP where the domain points to, I am unable to connect to the database and get an error 500. However, when I test and open all IPs in the MongoDB instance, I am able to connect.

I am currently using Vercel to host my app. 

Is there a separate IP address I should be whitelisting?

### Steps to Reproduce

My app uses MongoDB for its backend, and when I whitelist the IP where the domain points to, I am unable to connect to the database. However, when I test and open all IPs in the MongoDB instance, I am able to connect.

**Comment by amyegan** [comment]
Hey @slayt12. Because of the dynamic nature of Vercel deployments, there is no fixed IP to whitelist. Currently, you need to allow all IP addresses on your backend cloud and secure it in another way if necessary.

With the addition of [Vercel Secure Compute](https://vercel.com/docs/security/secure-compute#), Enterprise teams have the option to put their deployments and build container in a private network with dedicated IP addresses. But that is an Enterprise-only feature. I know most folks are on either Hobby or Pro plans, but I'm not sure which plan you're on and didn't want to leave this out in case it helps you: https://vercel.com/guides/how-to-allowlist-deployment-ip-address
